src/odemis/dataio/test/tiff_test.py

```diff
@@ -2729,6 +2732,77 @@ def test_convert_openfibsem_to_odemis_metadata(self):
             else:
                 self.assertEqual(value, data.content[0].metadata[key])
 
+PYRAMID_FILENAME = "test-pyramid.ome.tiff"
+class TestTileRead(unittest.TestCase):
+
+    def tearDown(self):
+        # clean up
+        try:
+            os.remove(PYRAMID_FILENAME)
+        except Exception:
+            pass
+
+    def test_tiled_read_and_projection(self):
+        """Test the tiled read functions"""
+
+        # storing full metadata for completeness
+        md = {
+            model.MD_PIXEL_SIZE: (7.738095238095199e-08, 7.738095238095199e-08),
+            model.MD_POS: (0.051781573195293, -0.007091270569792),
+            model.MD_ACQ_DATE: 1727296375,
+            model.MD_ROTATION: 6.1837053071795856,
+            model.MD_SHEAR: -4.3021142204224836e-16,
+            model.MD_DIMS: "YX",
+        }
+
+        # create a dummy overview pyramid with the same problematic shape
+        das = model.DataArray(
+            numpy.zeros((8194, 7143), dtype=numpy.uint16), metadata=md
+        )
+
+        # export the dummy overview pyramid
+        export(PYRAMID_FILENAME, das, pyramid=True)
```

Comment:
Is this just temporarily stored next to the actual .py file or root? Maybe it's neater to work with a temp folder (tempfile.TemporaryDirectory) that is instantiated in a setup class. Note that I'm not familiar with the I/O of the used functions, so maybe it's not trivial to implement if they don't accept file paths.

Reply:
It's a temp file stored in the same directory. It's removed when the class exits (tearDown)

Reply:
I get it, but it just doesn't look as safe as using an actual temp file or dir via said library, which lives in the os's tmp dir in a randomly named dirname (thus works with multiple parallel instances of the same or similar test). It probably works "fine", but utilizing actual temp file utility is safer in terms of isolation, automatic cleanup, and concurrency.